core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+
+/**
+ * Created by @author Aditya-Sood (21/05/19) as a part of GSoC 2019
+ *
+ * AddNoteDialog extends DialogFragment and is used to display the note corresponding to a
+ * particular article (of a particular zim file/wiki/book) as a full-screen dialog fragment.
+ *
+ * Notes are saved as text files at location: "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+ */
+
+// constant
+const val DISABLE_ICON_ITEM_ALPHA = 130
+const val ENABLE_ICON_ITEM_ALPHA = 255
+
+class AddNoteDialog : DialogFragment() {
+  private var zimFileName: String? = null
+  private var zimFileTitle: String? = null
+  private var articleTitle: String? = null
+
+  // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  private lateinit var articleNoteFileName: String
+  private var noteFileExists = false
+  private var noteEdited = false
+
+  private lateinit var root: View
+
+  // Keeps track of state of the note (whether edited since last save)
+  // Stores path to directory for the currently open zim's notes
+  private var zimNotesDirectory: String? = null
+
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @JvmField @Inject
+  var alertDialogShower: AlertDialogShower? = null
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    coreComponent
+      .activityComponentBuilder()
+      .activity(requireActivity())
+      .build()
+      .inject(this)
+
+    // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+    zimFileName = zimReaderContainer.zimCanonicalPath
+    if (zimFileName != null) { // No zim file currently opened
+      zimFileTitle = zimReaderContainer.zimFileTitle
+      articleTitle = (activity as WebViewProvider?)?.getCurrentWebView()?.title
+
+      // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+      val zimNoteDirectoryName = zimNoteDirectoryName
+      articleNoteFileName = getArticleNotefileName()
+      zimNotesDirectory = "$NOTES_DIRECTORY$zimNoteDirectoryName/"
+    } else {
+      onFailureToCreateAddNoteDialog()
+    }
+  }
+
+  private fun onFailureToCreateAddNoteDialog() {
+    context.toast(R.string.error_file_not_found, Toast.LENGTH_LONG)
+    closeKeyboard()
+    requireFragmentManager().beginTransaction().remove(this).commit()
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    super.onCreateView(inflater, container, savedInstanceState)
+    root = inflater.inflate(R.layout.dialog_add_note, container, false)
+    return root
+  }
+
+  private val zimNoteDirectoryName: String
+    get() {
+      val noteDirectoryName = getTextAfterLastSlashWithoutExtension(zimFileName!!)
+      return (if (noteDirectoryName.isNotEmpty()) noteDirectoryName else zimFileTitle)!!
```

Comment:
please don't use the `!!` operator

Reply:
@rashiq What should be the default value of zimFileName and ZimFileTitle?